src/adagrams.js

```diff
@@ -1,15 +1,124 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {...LETTER_POOL};
+  const TenStringsArray = []
+  for(let i = 0; i < 10; i++) {
+    const availableLetters = Object.keys(letterPool).filter(
+      (letter) => letterPool[letter] > 0
+  );
+    const randomIndex = Math.floor(Math.random()*availableLetters.length);
+    const chosenLetter = availableLetters[randomIndex];
+    TenStringsArray.push(chosenLetter);
+
+    letterPool[chosenLetter] -= 1;
+  }
+  return TenStringsArray;
 };
 
+
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const upperInput = input.toUpperCase()
+  const copyLettersInHand = [...lettersInHand]
+  for (const char of upperInput){
+    const index = copyLettersInHand.indexOf(char);
+    if (index === -1) {
+      return false;
+    }
+    copyLettersInHand.splice(index, 1);
```

Comment:
These are inputs are going to be the same size (10) but what if they weren't? What if the input size varied? What would be the time complexity of your code? What CS fundamentals data structure did we learn about that is handy for keeping track of the number of occurrences?